processor/src/host/default.rs

```diff
@@ -80,21 +80,21 @@ where
     /// `fn(&mut ProcessState) -> Result<(), EventHandler>`
     pub fn load_handler(
         &mut self,
-        id: u32,
-        handler: impl EventHandler,
+        id: Felt,
+        handler: Arc<dyn EventHandler>,
     ) -> Result<(), ExecutionError> {
-        self.event_handlers.register(id, Box::new(handler))
+        self.event_handlers.register(id, handler)
     }
 
     /// Unload a handler with the given id, returning a flag indicating whether a handler
     /// was previously registered with this id.
-    pub fn unload_handler(&mut self, id: u32) -> bool {
+    pub fn unload_handler(&mut self, id: Felt) -> bool {
         self.event_handlers.unregister(id)
     }
```

Comment:
nit: should we stick with the `register`/`unregister` convention, instead of using both that and `load/unload`?